fdk.go

```diff
@@ -175,12 +176,16 @@ func doJSONOnce(handler Handler, ctx context.Context, in io.Reader, out io.Write
 	json.NewEncoder(out).Encode(jsonResponse)
 }
 
-func doHTTPOnce(handler Handler, ctx context.Context, in io.Reader, out io.Writer, buf *bytes.Buffer, hdr http.Header) {
-	// TODO we need to set deadline on ctx here (need FN_DEADLINE header)
-	// for now, just get a new ctx each go round
-	ctx, cancel := context.WithCancel(ctx)
-	defer cancel()
+func CtxWithDeadline(parent context.Context, fnDeadline string) (context.Context, context.CancelFunc) {
```

Comment:
can we lower case this? don't really want it to pollute public API in docs, we're giving them a handle on a context already, they shouldn't need this.